xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubPersistenceProviderManager.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2019 Ignite Realtime Foundation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.openfire.pubsub;
+
+import org.jivesoftware.util.ClassUtils;
+import org.jivesoftware.util.JiveGlobals;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Manages the persistence provider used to manage pubsub data in persistent storage.
+ *
+ * @author Guus der Kinderen, guus.der.kinderen@gmail.com
+ */
+public class PubSubPersistenceProviderManager
+{
+    private static final Logger Log = LoggerFactory.getLogger( PubSubPersistenceProviderManager.class );
+
+    private static PubSubPersistenceProviderManager instance;
+
+    private PubSubPersistenceProvider provider;
+
+    public synchronized static PubSubPersistenceProviderManager getInstance()
```

Comment:
If getInstance() is going to be called a lot, is it worth double-checking rather then using synchronized on the whole method?

i.e. 

```
public static PubSubPersistenceProviderManager getInstance() {
    if ( instance == null ) {
        synchronized(PubSubPersistenceProviderManager.class) {   
            if ( instance == null ) {
                instance = new PubSubPersistenceProviderManager();
            }
    }
    return instance;
}
```

Reply:
I'm not particularly fond of these constructs. They make the code more complex (novice devs will make weird faces when looking at this). My experience with "things like this" in the broadest sense is that the JVM optimizes things in ways that make changes like this mostly ineffective anyways. Unless there's a strong indication that this has a measurable impact on performance, I'd prefer to not prematurely optimize.